code/datums/loadout/loadout_manager.dm

```diff
@@ -4,32 +4,25 @@
  */
 /datum/loadout_manager
 	/**
-	 * List of all names/jobs of all loadouts. 
-	 * The format is like this: 
-	 * list(loadout_job_a, loadout_name_a, loadout_job_b, loadout_name_b,.... etc)
-	 * This is because jatum doesn't like list of list
-	 * This is converted to a list of list when sending to tgui
+	 * List of all loadouts. Format is list(list(loadout_job, loadout_name))
 	 */
 	var/loadouts_data = list()
 	/// The host of the loadout_manager, aka from which loadout vendor are you managing loadouts
-	var/loadout_vendor 
+	var/loadout_vendor
 	/// The version of the loadout manager
 	var/version = CURRENT_LOADOUT_VERSION
 
 ///Remove the data of a loadout from the loadouts list
 /datum/loadout_manager/proc/delete_loadout(mob/user, loadout_name, loadout_job)
-	for(var/i = 1; i <= length(loadouts_data); i += 2)
-		if(loadout_job == loadouts_data[i] && loadout_name == loadouts_data[i+1])
-			loadouts_data -= loadouts_data[i+1]
-			loadouts_data -= loadouts_data[i]
-			user.client.prefs.delete_loadout(loadout_name, loadout_job)
+	for(var/loadout_data in loadouts_data)
+		if(loadout_data[1] == loadout_job && loadout_data[2] == loadout_name)
+			loadouts_data -= loadout_data
```

Comment:
Does this not copy the list meaning you iterate over the same object multiple times

Reply:
I don't get what you mean.

All i can say is that this do delete one and only one loadout, and the right one